CriticalMass/APIRequest.swift

```diff
@@ -5,16 +5,24 @@ enum APIRequestBuildError: Error {
 }
 
 public protocol APIRequestDefining {
+    // String is just a default implementation and should be redefined if the Request uses query items
+    associatedtype QueryDefiningDataType: Encodable = String
     associatedtype ResponseDataType: Decodable
     var endpoint: Endpoint { get }
     var httpMethod: HTTPMethod { get }
     var headers: HTTPHeaders? { get }
-    var queryItems: [URLQueryItem]? { get }
+    var queryItem: QueryDefiningDataType? { get }
     var requiresBackgroundTask: Bool { get }
     func makeRequest() throws -> URLRequest
     func parseResponse(data: Data) throws -> ResponseDataType
 }
 
+extension APIRequestDefining {
+    var queryItem: QueryDefiningDataType? {
+        nil
```

Comment:
So no need to explicitly return that as a default

Reply:
Extensions only support computed properties so we have to explicitly return that. 

Having a default implementation in an extension in this case has two benefits
1. APIRequestDefining implementations that don't need a queryItem don't need to implement it
2. Implementations that need to define can override it with a specific type instead of Encodable

Reply:
But this does not need to have a default implementation in the extension since it is `nil` already

Reply:
Yes, but without that default implementation, each APIRequestDefining implementation has to implement `var queryItem: String?` which I'm trying to avoid. Especially since assigning a String doesn't really work because String is just a placeholder type. 
Can also remove that and add the implementations.

Reply:
True. Missed that.
Maybe a builder instead of the make request function in the protocol would make things clearer. But not for this pr

Reply:
Yes, also though about experimenting with compositing multiple protocols to eliminate optionals in a future PR